omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,208 @@
+		assertEquals(IDENTIFIER_TYPE_UUID, PropertyUtils.getProperty(newAutoGenerationOption, "identifierType"));
+		assertEquals(automaticGenerationEnabled,
+		    PropertyUtils.getProperty(newAutoGenerationOption, "automaticGenerationEnabled"));
+		assertEquals(manualEntryEnabled, PropertyUtils.getProperty(newAutoGenerationOption, "manualEntryEnabled"));
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOptionWithoutLocation() throws Exception {
+		long originalCount = getAllCount();
+		boolean automaticGenerationEnabled = true;
+		boolean manualEntryEnabled = true;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertEquals(SOURCE, PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertEquals(UNKNOWN_LOCATION, PropertyUtils.getProperty(newAutoGenerationOption, "location"));
+		assertEquals(IDENTIFIER_TYPE_UUID, PropertyUtils.getProperty(newAutoGenerationOption, "identifierType"));
+	}
+	
+	@Test(expected = org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException.class)
+	public void shouldNotCreateNewAutoGenerationOptionWithNullParameters() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", null);
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
+		autoGenerationOption.add("automaticGenerationEnabled", null);
+		autoGenerationOption.add("manualEntryEnabled", null);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		handle(req);
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+		assertEquals(LOCATION,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getLocation().getUuid());
+		assertEquals(SOURCE,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getSource().getUuid());
+		assertTrue(
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertTrue(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+		boolean automaticGenerationEnabled = false;
```

Comment:
What is the use of these two local variables? Can't you just pass in the actual values?